source/constructs/api/data_source/service.py

```diff
@@ -2290,18 +2299,17 @@ def __list_rds_schema(account, region, credentials, instance_name, payload, rds_
                 'SecurityGroupIds': rds_security_groups
             },
         )
-        state = ''
-        time_elapsed = 0
-        timeout = 600
-        while state != 'Active':
-            response = lambda_.get_function(FunctionName=function_name)
-            logger.info("__list_rds_schema get lambda function:")
-            state = response['Configuration']['State']
-            logger.info(state)
-            sleep(SLEEP_TIME)
-            time_elapsed += SLEEP_TIME
-            if time_elapsed >= timeout:
-                break
+    time_elapsed = 0
+    timeout = 600
+    while state != 'Active':
+        sleep(SLEEP_TIME)
+        time_elapsed += SLEEP_TIME
+        if time_elapsed >= timeout:
+            break
+        response = lambda_.get_function(FunctionName=function_name)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code appears to be waiting for a resource before it runs. You could use the waiters feature to help improve efficiency. Consider using function_exists. For more information, see [https://boto3.amazonaws.com/v1/documentation/api/latest/guide/clients.html#waiters](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/clients.html#waiters)